constellations/orion/src/repos/account_repo.go

```diff
@@ -117,32 +118,82 @@ func (acc *accountRepo) SelectAllNegativeBalances() ([]domains.AccountSum, error
 	return results, nil
 }
 
-func (acc *accountRepo) Insert(account domains.Account) error {
+func (acc *accountRepo) InsertWithUser(account domains.Account, user domains.User) error {
+	tx, err := acc.db.Begin()
+	if err != nil {
+		return err
+	}
 	utils.LogWithContext("accountRepo.Insert", logger.Fields{"account": account})
 	statement := "INSERT INTO accounts (" +
 		"created_at, " +
 		"updated_at, " +
 		"primary_email, " +
 		"password" +
 		") VALUES (?, ?, ?, ?)"
-
 	stmt, err := acc.db.Prepare(statement)
 	if err != nil {
+		tx.Rollback()
 		return appErrors.WrapDbPrepare(err, statement)
 	}
 	defer stmt.Close()
 
 	now := time.Now().UTC()
-	execResult, err := stmt.Exec(
+	_, err = stmt.Exec(
 		now,
 		now,
 		account.PrimaryEmail,
 		account.Password,
 	)
 	if err != nil {
-		return appErrors.WrapDbExec(err, statement, account)
+		tx.Rollback()
+		return err
+	}
```

Comment:
add another if statement here.

We should also actually validate the Db Result. This part might be a bit confusing. As you might've noticed LOTS of things can go wrong with SQL / MySQL. 

statement.Exec() provides a "Result" struct or an error. The "Result" struct has information about how many rows were updated for example. Here, we want to check at least 1 row gets updated in the account table.

So I would do something like this:
```
result1, err := stmt.Exec(...)

if err != nil {
		tx.Rollback()
		return appErrors.WrapExec(err)
}
if err = appErrors.ValidateDbResult(result1, 1, "account was not inserted"); err != nil {
    tx.Rollback()
    return appErrors.WrapDbExec(err, statement, account)
}
```

this is an exception (you don't have to do this for stmt.Query(...)), but here two things can go wrong. The first if handles the `err` case. The second handles the sql.Result struct that might have inconsistent information. I do all of that for you in the appErrors package.